recruitmentCog/recruitment.py

```diff
@@ -0,0 +1,226 @@
+        if not isinstance(ctx.author, discord.Member):
+            await ctx.send("You need to join the server before your application can be processed.")
+            return False
+        if not isinstance(ctx.channel, discord.DMChannel):
+            try:
+                await ctx.message.delete()
+            except:
+                pass
+            await self.interactive_application(ctx)
+            return False
+        return True
+
+
+    async def interactive_application(self, ctx: commands.Context):
+        """Ask the user several questions to create an application."""
+        author = ctx.author
+        embed = discord.Embed(
+            title="+++ KANIUM APPLICATION SYSTEM +++",
+            description="Ah, you wish to apply for Kanium membership. Very well, understand that This process is very important to us so we expect you to put effort in and be glorious about it. Let us begin!",
+            color=discord.Color.green(),
+        )
+        await author.send(embed=embed)
+
+        answers = await self.ask_questions(author)
+
+        if not answers:
+            return
+
+        embeddedApplication = await self.format_application(answers, author)
+        
+        # Call the sendApplication to check if the author is a member of the guild and send the application if they are
+        await self.sendApplication(author, embeddedApplication)
+
+
+    async def sendApplication(self, author: discord.Member, embeddedApplication: discord.Embed):
+        # Check if the author is a member of the guild
+        guild = author.guild
+        member = guild.get_member(author.id)
+        if member is None:
+            await author.send("You need to join the server before your application can be processed.")
+            return
+
+        # Send the embed to the application channel
+        application_channel_id = await self.config.guild(guild).application_channel_id()
+        if not application_channel_id:
+            await author.send("The application channel has not been set. Please use the `setapplicationschannel` command to set it.")
+            return
+
+        application_channel = guild.get_channel(application_channel_id)
+        if application_channel is None:
+            await author.send(f"The application channel with ID {application_channel_id} could not be found.")
+            return
+
+        try:
+            message = await application_channel.send(embed=embeddedApplication)
+        except discord.Forbidden:
+            await author.send("I do not have permission to send messages in the application channel.")
+            return
+
+        # Add reactions to the message
+        try:
+            await self.add_reactions(message)
+        except discord.Forbidden:
+            await author.send("I do not have permission to add reactions to messages in the application channel.")
+            return
+
+        # Assign the Trial role to the author
+        role = guild.get_role(531181363420987423)
```

Comment:
Same as above, hardcoded roles, ids, etc should be in an env variable config as it can change depending on the usage

Reply:
Done